nbodykit/dataset.py

```diff
@@ -0,0 +1,849 @@
+            a dictionary holding the data variables, where the keys
+            are interpreted as the variable names
+            
+        force_index_match : bool, optional
+             if `True`, when indexing using coordinate values, 
+             automatically return the results for the closest coordinate
+             to the input values
+             
+        sum_only : list
+            a list of strings specifying variables that will 
+            only be summed (not averaged) when combining bins
+            
+        **kwargs :
+            any additional keywords are saved as metadata in the ``attrs``
+            attribute of the class
+        """
+        if len(dims) != len(edges):
+            raise ValueError("size mismatch between specified `dims` and `edges`")
+        
+        shape = tuple(len(e)-1 for e in edges)
+        for name in variables:
+            if numpy.shape(variables[name]) != shape:
+                args = (shape, numpy.shape(variables[name]))
+                raise ValueError("`edges` imply data shape of %s, but data has shape %s" %args)
+        
+        self.dims = [dim+'_cen' for dim in dims]
+        self.edges = dict(zip(self.dims, edges))
+        
+        # coordinates are the bin centers
+        self.coords = {}
+        for i, dim in enumerate(self.dims):
+            self.coords[dim] = 0.5 * (edges[i][1:] + edges[i][:-1])
+            
+        # store variables as a structured array
+        dtypes = [(name, variables[name].dtype) for name in variables]            
+        self.data = numpy.empty(self.shape, dtype=numpy.dtype(dtypes))
+        for name in variables:
+            self.data[name] = variables[name]
+            
+        # define a mask such that a coordinate grid element will be masked
+        # if any of the variables at that coordinate are (NaN, inf)
+        self.mask = numpy.zeros(self.shape, dtype=bool)
+        for name in variables:
+            self.mask = numpy.logical_or(self.mask, ~numpy.isfinite(self.data[name]))
+             
+        # If `True`: always returns nearest bin value
+        self.force_index_match = force_index_match
+        
+        # variables which are not averaged
+        self.sum_only = sum_only
+        
+        # save track metadata
+        self.attrs = OrderedDict()
+        for k in kwargs:
+            self.attrs[k] = kwargs[k]
+            
+    @property
+    def ndim(self):
+        """
+        The number of coordinate dimensions
+        """
+        return len(self.dims)
+        
+    @property
+    def shape(self):
+        """
+        The shape of the coordinate grid
+        """
+        return tuple(len(self.coords[d]) for d in self.dims)
+        
+    @property
+    def variables(self):
+        """
+        Alias to return the names of the variables stored in `data`
+        """
+        return list(self.data.dtype.names)
+        
+    @classmethod
+    def __construct_direct__(cls, data, mask, **kwargs):
```

Comment:
Yes. I know it is for internal use only... but just by looking at the code I can only tell when it used, but cannot easily tell the difference between the object it returns with an object created by DataSet(....). I worry after a month you'll get confused too..